docs/minikube.md

```diff
@@ -0,0 +1,49 @@
+# Minikube
+
+This document details Minikube specific deployment instructions.
+Minikube is not recommended as a platform to test or develop Contour due to its network limitations.
+
+## Access your cluster
+
+On Minikube you can retrieve the external address of Contour's load balancer:
+```
+$ minikube service -n heptio-contour contour --url
+http://192.168.99.100:30588
+```
+
+## Configuring DNS
+
+On Minikube, you can fake DNS by editing `/etc/hosts` or you can use the provided example and not have to modify dns on your local machine.
+
+## Deploy KUARD example application.
+
+```
+$ kubectl apply -f https://j.hept.io/contour-kuard-minikube-example
+```
+
+This example yaml specifies `kuard.192.168.99.100.nip.io` as a specific ingress backend for kuard.
+It uses nip.io and the minikube ip address to have kuard only respond to http://kuard.192.168.99.100.nip.io.
+Once that is applied you can visit http://kuard.192.168.99.100.nip.io and see the kuard example application.
+
+## Troubleshooting
+
+Minikube remaps Contour's service loadbalancer from ports 80 and 443 to a random high port as the ingress.
+This is problematic because this port is not a _well known_ (see RFC 2616) port so `curl(1)` or browsers will include the port number in the `Host:` header.
```

Comment:
"This is not a well-known port, which means that `curl(1)` or a browser includes the port number in the `Host:` header. (See RFC 2616 for more information.)"

Q: I did a quick rummage through the RFC and couldn't see any obviously useful information there. There's no explicit documentation of well-known ports, nor anything immediately obvious about the Host: header and ports being included. I understand what the RFC is saying and how it maps to what you're saying here, but I'm not convinced it's going to help a Minikube user in this context. I was going to recommend an explicit link, but then decided not to. If you agree with my assessment, let's take out any mention of the RFC altogether. Otherwise, as ^ in my edit is OK."

Reply:
Yeah, the underlying issue is the RFC doesn't say if you should include a port in your Host: header always, sometimes (ie, when it's not a well known port), or never. Hence every web server makes up its own rules about how to match Hosts to routes.